rechunker/core.py

```diff
@@ -0,0 +1,17 @@
+from typing import Iterable, NamedTuple, Tuple, Union
+
+import dask.array
+import zarr
+
+
+class CopySpec(NamedTuple):
+    source: Union[zarr.Array, dask.array.Array]
```

Comment:
It would be nice to avoid the dask dependency here while preserving type safety, but it's needed after https://github.com/pangeo-data/rechunker/pull/27. Perhaps these could use a Generic or a Protocol?

Reply:
I totally open to changing things around! I'd like to be able to accept Dask arrays as `source`, at least when executing with Dask, because this will probably be a common workflow (compute something with one chunk structure and then store it with a different one). Is there a way that the staged_copy implementation could somehow register what sort of sources it supports?

Reply:
This isn't hard to resolve -- the supported source array types can definitely be backend specific. It's a little annoying to type annotate but that's really a second order concern :). For now I'll just use `Any`.